### PR TITLE
refactor / Refactor arbitrage strategy for easier unit testing

### PR DESCRIPTION
**Pull request by michaelc9**
**Before submitting this PR, please make sure**:
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

**Optional**:
- Related Github issue:
- Related Clubhouse Story:

**A description of the changes proposed in the pull request**:

### CODE REVIEW

File: hummingbot/strategy/arbitrage/arbitrage.pxd [hunk 1]
```diff
@@ -37,7 +37,7 @@ cdef class ArbitrageStrategy(StrategyBase):
     cdef c_did_complete_sell_order(self, object sell_order_completed_event)
     cdef c_did_fail_order(self, object fail_event)
     cdef c_did_cancel_order(self, object cancel_event)
-    cdef tuple c_calculate_arbitrage_profitability(self,
+    cdef tuple c_calculate_arbitrage_top_order_profitability(self,
                                                    object market_pair,
```
Comment (chowryan) [inline hummingbot/strategy/arbitrage/arbitrage.pxd]: fix spacing under this line

File: hummingbot/strategy/arbitrage/arbitrage.pyx [hunk 2]
```diff
@@ -95,7 +95,6 @@ cdef class ArbitrageStrategy(StrategyBase):
         self._last_timestamp = 0
         self._next_trade_delay = next_trade_delay_interval
         self._last_trade_timestamps = {}
-        self._exchange_rate_conversion = ExchangeRateConversion.get_instance()
```
Comment (chowryan) [inline hummingbot/strategy/arbitrage/arbitrage.pyx]: why was this removed/refactored?

File: hummingbot/strategy/arbitrage/arbitrage.pyx [hunk 3]
```diff
@@ -333,45 +337,73 @@ cdef class ArbitrageStrategy(StrategyBase):
             if market_pair in self._tracked_market_orders:
                 del self._tracked_market_orders[market_pair]
 
-    cdef tuple c_calculate_arbitrage_profitability(self,
+    cdef tuple c_calculate_arbitrage_top_order_profitability(self,
```
Comment (chowryan) [inline hummingbot/strategy/arbitrage/arbitrage.pyx]: fix spacing below

File: hummingbot/strategy/arbitrage/arbitrage.pyx (hunk 2 above)
Comment (michaelc9) [reply hummingbot/strategy/arbitrage/arbitrage.pyx]: It is a singleton class and we don't need an instance stored in the strategy instance. One of the strategy method is moved outside of the strategy class to be re-used in discovery strategy.
